### PR TITLE
[core] Add object owner and copy metrics to node stats

### PR DESCRIPTION
**Pull request by fishbone**
## Why are these changes needed?

This PR adds the object owner and copy metrics to `GetNodeStats` RPC endpoint.

The inlined small objects are not counted as one copy because it's not stored in object store and when it's used, it'll be copied inline, so no need to count it.

But it's still counted as 1 as ownership for correctness because it's actually owned by worker.

The local copies are retrieved from local object manager directly and owner counts needs the caller to aggregate the metrics from each core worker.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jjyao** [comment]
How will those stats be used?

**Comment by iycheng** [comment]
@jjyao Lanbo will use this to observe whether a node can't be downscale is due to object.

**Review by scv119 — APPROVED** [review]

File: src/ray/protobuf/node_manager.proto [hunk 1]
```diff
@@ -219,6 +219,8 @@ message ObjectStoreStats {
   // the node has more pull requests than available object store
   // memory.
   bool object_pulls_queued = 13;
+  // The number of primary copies of objects in the local node.
+  int64 num_object_store_primary_copies = 14;
```
Comment (lanbochen-anyscale) [inline src/ray/protobuf/node_manager.proto:223]: So this number is for objects owned by this raylet and the above number is for objects owned by individual worker?

Comment (iycheng) [reply src/ray/protobuf/node_manager.proto:223]: @lanbochen-anyscale  

> this number is for objects owned by this raylet

primary copy of the objects in this raylet

> number is for objects owned by individual worker

Yes, you need to aggregrate it in cli side.

**Pull request merged**